lib/measures/urban_geometry_creation/measure.rb

```diff
@@ -52,657 +53,32 @@ def description
-        vector = other_building_point - building_point
-        all_pairs << {:building_point => building_point, :other_building_point => other_building_point, :vector => vector, :distance => vector.length}
-      end
-    end
-    
-    all_pairs.sort! {|x, y| x[:distance] <=> y[:distance]}
-    
-    all_pairs.each do |pair|
-      if point_is_shadowed(pair[:building_point], pair[:other_building_point])
-        return true
-      end
-    end
-    
-    return false
-  end
-  
-  
-  def point_is_shadowed(building_point, other_building_point)
-  
-    vector = other_building_point - building_point
-
-    height = vector.z
-    distance = Math.sqrt(vector.x*vector.x + vector.y*vector.y)
-    
-    if distance < 1
-      return true
-    end
-    
-    hour_angle_rad = Math.atan2(-vector.x, -vector.y)
-    hour_angle = OpenStudio::radToDeg(hour_angle_rad)
-    lattitude_rad = OpenStudio::degToRad(@origin_lat_lon.lat)
-    
-    result = false
-    (-24..24).each do |declination|
-      
-      declination_rad = OpenStudio::degToRad(declination)
-      zenith_angle_rad = Math.acos(Math.sin(lattitude_rad)*Math.sin(declination_rad) + Math.cos(lattitude_rad)*Math.cos(declination_rad)*Math.cos(hour_angle_rad))
-      zenith_angle = OpenStudio::radToDeg(zenith_angle_rad)
-      elevation_angle = 90-zenith_angle
-        
-      apparent_angle_rad = Math.atan2(height, distance)
-      apparent_angle = OpenStudio::radToDeg(apparent_angle_rad)
-    
-      #puts "declination is #{declination}.  Other building is #{distance} m away and #{height} m high, hour_angle is #{hour_angle}, apparent angle is #{apparent_angle}, zenith_angle is #{zenith_angle}, elevation_angle is #{elevation_angle}"
-      
-      if (elevation_angle > 0 && elevation_angle < apparent_angle)
-        result = true
-        break
-      end
-    end
-    
-    #puts "lattitude = #{@origin_lat_lon.lat}, vector = #{vector}, distance = #{distance}, height = #{height}, shadowed = #{result}"
-    
-    return result
-  end
-  
-  # get the feature from the geojson
-  def get_feature(feature_id)
-    
-    puts feature_id
-    @geojson[:features].each do |f|
-      if f[:properties] && f[:properties][:source_id] == feature_id
-        return f
-      end
-    end
-    
-    return nil
-  end
-  
-  # get the feature collection from the database
-  def get_feature_collection(params)
-    
-    #params[:commit] = 'Proximity Search'
-    #params[:feature_id] = feature_id
-    #params[:distance] = 100
-    #params[:proximity_feature_types] = ['Building']
- 
-    return {}
-  end
-  
   # define what happens when the measure is run
   def run(model, runner, user_arguments)
     super(model, runner, user_arguments)
-
+    geojson_gem = URBANopt::GeoJSON::GeoJSON.new
     # use the built-in error checking
-    if !runner.validateUserArguments(arguments(model), user_arguments)
+    if !runner.validateUserArguments(geojson_gem.arguments(model), user_arguments)
```

Comment:
@karinamzalez `arguments` is a method of the measure like `run`, http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/#measure-elements, it shouldn't be a method of the geojson_gem